### PR TITLE
Changes the targeting code for hostile animals

### LINKED ISSUE #9664: hostile mobs that have been turned sentient will turn back hostile if controlling player disconnects
What the issue title says. If you apply a sentience potion to a hostile mob(for example, everything spawned by the chem life reaction) and a player accepts, you'll get a nice, friendly human controlled mob that obeys your commands. But when said player disconnects, the mob will turn hostile again, possibly killing everyone around it.

### LINKED ISSUE #10570: Sentient xenobio aliens randomly plant weeds
Kind of weird. The apparent random weed planting code for NPC aliens continues working even when a player is in control of the mob. Player-controlled slime core aliens don't have access to the alien tab (i.e. they can't plant weeds normally).

### PR DESCRIPTION
**Pull request by phil235**
- to be a bit more efficient. AIStatus and stance vars are now only one var. The idle stance is fused with the AI_SLEEP status (which is renamed AI_IDLE). hostile/life() now only calls ListTargets() once per cycle (we save a list of targets then use that instead of calling the proc again). This will help with #9624.
- Alien animals no longer plant egg/weed when player controlled (even if disconnected). Fixes #10570
  Created the proc handle_automated_action() for simple_animals , currently used by hostile animal targeting,  by giant spiders (cocoon stuff) and alien animals (weed planting). 
- Player controlled hostile animals no longer starts targetting things again when the player disconnects. Fixes #9664

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -36,39 +35,35 @@
...
-
-		if(AIShouldSleep())
-			AIStatus = AI_SLEEP
 
 
 //////////////HOSTILE MOB TARGETTING AND AGGRESSION////////////
 
 
 /mob/living/simple_animal/hostile/proc/ListTargets()//Step 1, find out what we can see
+	world << "DEBUG: ListTargets()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -81,19 +76,19 @@
 		L += Objects
 	return L
 
-/mob/living/simple_animal/hostile/proc/FindTarget()//Step 2, filter down possible targets to things we actually care about
+/mob/living/simple_animal/hostile/proc/FindTarget(var/list/possible_targets, var/HasTargetsList = 0)//Step 2, filter down possible targets to things we actually care about
+	world << "DEBUG: FindTarget()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 3]
```diff
@@ -101,18 +96,19 @@
...
 			var/possible_target_distance = get_dist(src, A)
 			if(target_dist < possible_target_distance)
 				Targets -= A
-	if(!Targets.len)//We didnt find nothin!
-		return
 	var/chosen_target = pick(Targets)//Pick the remaining targets (if any) at random
 	return chosen_target
 
 /mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target?
+	world << "DEBUG: CanAttack()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 4]
```diff
@@ -150,15 +146,15 @@
 	target = new_target
 	if(target != null)
 		Aggro()
-		stance = HOSTILE_STANCE_ATTACK
-	return
+		return 1
 
-/mob/living/simple_animal/hostile/proc/MoveToTarget()//Step 5, handle movement between us and our target
+/mob/living/simple_animal/hostile/proc/MoveToTarget(var/list/possible_targets)//Step 5, handle movement between us and our target
+	world << "DEBUG: MoveToTarget()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 5]
```diff
@@ -229,7 +212,7 @@
 	taunt_chance = initial(taunt_chance)
 
 /mob/living/simple_animal/hostile/proc/LoseTarget()
-	stance = HOSTILE_STANCE_IDLE
+	world << "DEBUG: LoseTarget()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 6]
```diff
@@ -309,6 +292,7 @@
 	return
 
 /mob/living/simple_animal/hostile/proc/FindHidden()
+	world << "DEBUG: FindHidden()"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: debug

**Comment by Anonus** [comment]
phil continues to be amazing

**Pull request merged**